Core/src/main/java/com/paypal/android/core/graphql/common/Query.kt

```diff
@@ -0,0 +1,33 @@
+package com.paypal.android.core.graphql.common
+
+import org.json.JSONObject
+
+internal abstract class Query<T> {
+    fun requestBody(): JSONObject {
+        val query = """query { $queryName(${queryParameters()})
+            $dataFieldsForResponse }
+        """
+        return JSONObject().apply {
+            put("query", query)
+        }
+    }
+
+    abstract val queryParams: Map<String, Any>
+    abstract val queryName: String
+    abstract val dataFieldsForResponse: String
+    abstract fun parse(jsonObject: JSONObject): T
+
+    private fun queryParameters(): String {
+        return StringBuilder().apply {
+            for (pair in queryParams) {
+                append("\n ${pair.key}: ${pair.value.encodedValue()} \n ")
```

Comment:
question: Could we return a `JSONObject` here instead to avoid having to generate JSON key pairs?

Reply:
[GraphQL POST](https://graphql.org/learn/serving-over-http/#post-request) requests do support a `variables` field. It'll be less stress on the `Query` class if we don't have to think about encoding.